src/t_set.c

```diff
@@ -1093,8 +1093,13 @@ void srandmemberWithCountCommand(client *c) {
      * Listpack encoded sets are meant to be relatively small, so
      * SRANDMEMBER_SUB_STRATEGY_MUL isn't necessary and we rather not make
      * copies of the entries. Instead, we emit them directly to the output
-     * buffer. */
-    if (set->encoding == OBJ_ENCODING_LISTPACK) {
+     * buffer.
+     *
+     * This is actually an optimization for later CASE 4. Note that we don't
+     * do this in CASE 3 because we also expect a random order too. */
+    if (set->encoding == OBJ_ENCODING_LISTPACK &&
+        (count*SRANDMEMBER_SUB_STRATEGY_MUL <= size))
```

Comment:
So the order is only important if we're selecting more than 1/3 of the size of the set?

Reply:
ohh i forgot post the history
https://github.com/redis/redis/pull/8219#issuecomment-749587323
https://github.com/redis/redis/pull/8444

Reply:
OK, if this is what we wanted before, maybe we want it now as well. :-) I would let @oranagra decide.